api/src/jmh/java/io/opentelemetry/trace/propagation/BinaryTraceContextFromByteArrayBenchmark.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Thread)
+public class BinaryTraceContextFromByteArrayBenchmark {
+
+  private static final byte SAMPLED_TRACE_OPTIONS_BYTES = 1;
+  private final BinaryTraceContext binaryTraceContext = new BinaryTraceContext();
+  private final TraceFlags sampledTraceOptions = TraceFlags.fromByte(SAMPLED_TRACE_OPTIONS_BYTES);
+  private final TraceState traceStateDefault = TraceState.builder().build();
+  private final List<byte[]> byteArrays =
+      Arrays.asList(
+          createByteArray("905734c59b913b4a905734c59b913b4a", "9909983295041501"),
+          createByteArray("21196a77f299580e21196a77f299580e", "993a97ee3691eb26"),
+          createByteArray("2e7d0ad2390617702e7d0ad239061770", "d49582a2de984b86"),
+          createByteArray("905734c59b913b4a905734c59b913b4a", "776ff807b787538a"),
+          createByteArray("68ec932c33b3f2ee68ec932c33b3f2ee", "68ec932c33b3f2ee"));
+  private byte[] byteArray = byteArrays.get(0);
+  private Integer iteration = 0;
+
+  @Benchmark
+  @BenchmarkMode({Mode.Throughput, Mode.AverageTime})
+  @Fork(1)
+  @Warmup(iterations = 5, time = 1)
+  @Measurement(iterations = 50_000, time = 1, timeUnit = TimeUnit.MILLISECONDS)
```

Comment:
Why running only for a single millisecond? Feels like that is so short that it could cause unstable results?

Reply:
![74876720-3ee9a500-5375-11ea-9554-0ef40106d4de](https://user-images.githubusercontent.com/11577813/75054219-bb01fb00-54e3-11ea-9e53-c66f962464a7.png)
it do a tons of ops/s no need to run it for more

Reply:
From a readability standpoint of output data I would probably run longer iterations of a second and run 20-30 of them instead.

To make the output a bit more readable you can set the time unit used for the output to be milli, micro or nano instead will make it easier to read instead of getting the scientific notation.

Reply:
We should decrease a number of iteration but increase time of runtime, good idea